src/builder/build_traits.rs

```diff
@@ -37,9 +37,9 @@ use crate::hugr::HugrMut;
 pub trait Container {
     /// The container node.
     fn container_node(&self) -> Node;
-    /// The underlying [`HugrMut`] being used to build the HUGR.
-    fn base(&mut self) -> &mut HugrMut;
-    /// Immutable reference to HUGR being built.
+    /// The underlying [`Hugr`] being built...TODO: should we just require `AsMut<Hugr>`?
+    fn base(&mut self) -> &mut Hugr;
+    /// Immutable reference to HUGR being built...TODO: should we just require `AsRef<Hugr>`? Or combine with previous?
     fn hugr(&self) -> &Hugr;
```

Comment:
It's probably nicer for the borrow checker to keep the separate methods (I'd vote for renaming `base` to `hugr_mut` though).

Requiring `Container : AsMut<Hugr> + AsRef<Hugr>` may be confusing for the user, as it'd expose all the low-level mutation methods along with the builder ones.

Reply:
agree with renaming, and yes, the separate method makes things simpler I believe.

Reply:
Yes, if you call `hugr_mut()` to get a mutable ref and then "upcast" to a immutable ref, the mutable borrow on the receiver of the `hugr_mut()` is presumably still live.

Also agreed that `Container: AsMut<Hugr> + AsRef<Hugr>` could be confusing.

So, renamed, and TODOs removed.

In time I guess these will return (im)mutable views onto subregions.